### PR TITLE
[RFC] eval.c: NULL and empty string bugs

### PR DESCRIPTION
**Pull request by splinterofchaos**
See: #2286

### CODE REVIEW

File: src/nvim/eval.c [hunk 1]
```diff
@@ -12882,6 +12885,7 @@ static void f_rpcstart(typval_T *argvars, typval_T *rettv)
   uint64_t channel_id = channel_from_job(argv);
 
   if (!channel_id) {
+on_error:
```
Comment (oni-link) [inline src/nvim/eval.c]: `channel_id` is not initialized when we jump to this point and use it later as return value.

Comment (splinterofchaos) [inline src/nvim/eval.c]: Oops, I put this commit in the wrong branch! Oh well, I thought it was close enough related anyway.

Good catch :+1: . fixed.

File: src/nvim/eval.c [hunk 2]
```diff
@@ -12867,13 +12867,17 @@ static void f_rpcstart(typval_T *argvars, typval_T *rettv)
...
+    return
+  }
   argv[0] = xstrdup((char *)argvars[0].vval.v_string);
 
   int i = 1;
   // Copy arguments to the vector
   if (argsl > 0) {
     for (listitem_T *arg = args->lv_first; arg != NULL; arg = arg->li_next) {
-      argv[i++] = xstrdup((char *)arg->li_tv.vval.v_string);
+      argv[i++] = xstrdup((char *) get_tv_string(&arg->li_tv));
```
Comment (oni-link) [inline src/nvim/eval.c]: Same problem in `jobstart()`, where we assume that argument strings never point to `NULL`.

Comment (splinterofchaos) [inline src/nvim/eval.c]: :+1: -- I found a few others, too. Assuming the tests pass, I will probably squash them all into one commit.

**Comment by splinterofchaos** [comment]
Added a commit to fix #2339.

**Comment by Pyrohh** [comment]
Regarding the commit messages, IIRC you need to put `closes #[issue number]`, as `fixes #[issue number]` doesn't work (might be wrong here...).

**Comment by splinterofchaos** [comment]
Github does respect `fixes #_`, and several other aliases. https://help.github.com/articles/closing-issues-via-commit-messages/

However, that page doesn't mention it, but I thought I remembered reading that "fixes" would only be respected if the author had the permissions to close issues.

**Comment by Pyrohh** [comment]
> Github does respect fixes #_, and several other aliases

Ah, my mistake. I remember being able to close issues with commits before I had permission to close them in general, but it seems what you did does indeed work. For me at least, the reference to your commits from this PR in other issues have a little marker next to them, as mentioned here:

https://github.com/blog/1506-closing-issues-via-pull-requests

**Comment by ZyX-I** [comment]
@splinterofchaos AFAIK it is not the commits’ author who needs to have permissions, but the person who merges them in. (*Not* when you need to close issue in the other repository obviously.) First time heard about closing issues via PRs though.

**Pull request merged**